pkg/xds/ingress/dataplane.go

```diff
@@ -3,21 +3,39 @@ package ingress
 import (
 	"context"
 	"reflect"
+	"sort"
 
 	mesh_proto "github.com/Kong/kuma/api/mesh/v1alpha1"
 	core_mesh "github.com/Kong/kuma/pkg/core/resources/apis/mesh"
 	"github.com/Kong/kuma/pkg/core/resources/manager"
+	"github.com/Kong/kuma/pkg/xds/envoy"
 )
 
-type serviceSet []*mesh_proto.Dataplane_Networking_Ingress_AvailableService
+// tag sets represent map from tags (encoded as string) to number of instances
+type tagSets map[string]int
```

Comment:
It is always positive and at the end you cast it to `uint32` probably it can be `uint32` from the very beginning